kin-base/src/main/java/kin/base/Transaction.java

```diff
@@ -206,11 +210,50 @@ public String toEnvelopeXdrBase64() {
     }
   }
 
+  /**
+   * Creates a <code>Transaction</code> instance from previously build <code>TransactionEnvelope</code>
+   * @param envelope Base-64 encoded <code>TransactionEnvelope</code>
+   * @return
+   * @throws IOException
+   */
+  public static Transaction fromEnvelopeXdr(String envelope) throws IOException {
+    TransactionEnvelope transactionEnvelope = TransactionEnvelope.decode(Util.createXdrDataInputStream(envelope));
+    return fromEnvelopeXdr(transactionEnvelope);
+  }
+
+  /**
+   * Creates a <code>Transaction</code> instance from previously build <code>TransactionEnvelope</code>
+   * @param envelope Base-64 encoded <code>TransactionEnvelope</code>
+   * @return
+   */
+  public static Transaction fromEnvelopeXdr(TransactionEnvelope envelope) {
+    kin.base.xdr.Transaction tx = envelope.getTx();
+    int mFee = tx.getFee().getUint32();
```

Comment:
As the fee is stroops, shouldn't it be a long?

Reply:
Again this is the code copied from them and in this case it is affect more classes so lets talk about when we meet.

Reply:
ok